File: sentimental-credit/credit.py
from cs50 import get_string

card = ' '
a = len(card)

while(a != 16 or a != 17 or a != 14 ):
    card = get_string("Number: ")
    a = len(card)
sum = 0
s1 = 0


for i in (a , 0 , -1):
    no = list.append(2 * int(i-1))
    nu = list.append(int(i))

for d in no:
    sum += (d)

for d in nu:
    s1 += (d)

total = sum + s1

if total % 10 == 0:
    if a == 15:
        print("AMEX")
    elif a == 13:
        print("Visa")
    elif a == 16:
        print("MASTERCARD")

else:
    print("INVALID")

File: credit/credit.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int r, r1, count = 0, sum = 0, sum1 = 0, c = 0, i = 0, length;
    long int num;
    long int a = get_long("Enter the credit card number");
    num = a;
    while (num > 0)
    {
        num = num / 10;
        count += 1;
    }
    length = count / 2;
    int h[length];
    while (a > 0)
    {
        r = a % 10;
        c++;
        if (c % 2 != 0)
        {
            h[i] = 2 * r;
            i++;
        }
        else
        {
            sum1 = sum1 + r;
        }
    }
    for (i = 0; i < count / 2; i++)
    {
        r1 = h[i] % 10;
        sum = sum + r1;
    }
    printf("%d", sum + sum1);
    return 0;
}
